spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/autoconfiguration/XsuaaResourceServerJwkAutoConfiguration.java

```diff
@@ -33,11 +38,14 @@
 @AutoConfigureBefore(OAuth2ResourceServerAutoConfiguration.class) // imports OAuth2ResourceServerJwtConfiguration which
 																	// specifies JwtDecoder
 public class XsuaaResourceServerJwkAutoConfiguration {
+	private final Logger logger = LoggerFactory.getLogger(getClass());
 
 	@Bean
 	@ConditionalOnBean(XsuaaServiceConfiguration.class)
+	@ConditionalOnMissingClass({ "reactor.core.publisher.Mono", "reactor.core.publisher.Flux" })
```

Comment:
Is the assumption behind this configuration that all projects using WebMVC do not have project reactor in their classpath?

There are valid scenarios where project reactor is used in WebMVC projects:
- Using `WebClient` instead of `RestTemplate`
- Reactor is used by a dependency (e.g. [cf-java-client](https://github.com/cloudfoundry/cf-java-client))

Reply:
Hi @simontb 
Yes, this was my intention. Good point!!!
Do you have a better Idea instead of Mono...

Reply:
Did you try [ConditionalOnWebApplication](https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/autoconfigure/condition/ConditionalOnWebApplication.Type.html) already? Although I've never used it, from the documentation this looks like what we need.

```suggestion
	@ConditionalOnWebApplication(type = Type.REACTIVE)
```

Reply:
Done -> thanks for the hint